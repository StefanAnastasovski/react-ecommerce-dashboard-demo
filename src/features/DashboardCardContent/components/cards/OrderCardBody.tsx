import React from "react";

import { Box } from "@mui/material";

import {
  FONT_WEIGHT_500,
  LINK_N_TEXT_FONT_SIZE,
  CARD_ORDER_BODY_COLOR_TEXT,
} from "../../../../data/constants";

import { DASHBOARD_API } from "../../../../data/dashboardConfig";

import Paragraph from "../../../../components/Typography/Paragraph";

const OrderCardBody = () => {
  const styledText = {
    color: CARD_ORDER_BODY_COLOR_TEXT,
  };
  const styledValues = {
    fontSize: LINK_N_TEXT_FONT_SIZE,
    fontWeight: FONT_WEIGHT_500,
  };

  const {
    orders: { orders_received, earnings },
  } = DASHBOARD_API;

  return (
    <>
      <Box sx={{ display: "flex", justifyContent: "space-between" }}>
        <Paragraph styleProps={styledText}>Orders received:</Paragraph>
        <Paragraph styleProps={styledValues}>{orders_received}</Paragraph>
      </Box>
      <Box sx={{ display: "flex", justifyContent: "space-between" }}>
        <Paragraph styleProps={styledText}>Earnings:</Paragraph>
        <Paragraph styleProps={styledValues}>â‚¬ {earnings}</Paragraph>
      </Box>
    </>
  );
};

export default OrderCardBody;
