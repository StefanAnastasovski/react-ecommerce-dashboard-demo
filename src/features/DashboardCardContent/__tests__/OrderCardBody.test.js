import React from "react";
import { shallow } from "enzyme";

import OrderCardBody from "../components/cards/OrderCardBody";

describe("<OrderCardBody />", () => {
  describe("render()", () => {
    it("Shallow rendering - MatchSnapshot", () => {
      const component = shallow(<OrderCardBody />);
      expect(component).toMatchSnapshot();
    });
  });
});
