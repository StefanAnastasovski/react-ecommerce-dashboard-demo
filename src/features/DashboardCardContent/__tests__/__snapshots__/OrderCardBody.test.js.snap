// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderCardBody /> render() Shallow rendering - MatchSnapshot 1`] = `ShallowWrapper {}`;
